bootstrap.py

```diff
@@ -45,206 +51,123 @@
 parser.add_argument('--gui', default=None,
                     help="GUI toolkit: pyqt5 (for PyQt5) or pyside2 "
                     "(for PySide2)")
-parser.add_argument('--show-console', action='store_true', default=False,
-                    help="(Deprecated) Does nothing, now the default behavior "
-                    "is to show the console")
 parser.add_argument('--hide-console', action='store_true', default=False,
                     help="Hide parent console window (Windows only)")
 parser.add_argument('--safe-mode', dest="safe_mode",
                     action='store_true', default=False,
                     help="Start Spyder with a clean configuration directory")
-parser.add_argument('--no-apport', action='store_true', default=False,
-                    help="Disable Apport exception hook (Ubuntu)")
 parser.add_argument('--debug', action='store_true',
                     default=False, help="Run Spyder in debug mode")
 parser.add_argument('--filter-log', default='',
                     help="Comma-separated module name hierarchies whose log "
                          "messages should be shown. e.g., "
                          "spyder.plugins.completion,spyder.plugins.editor")
+parser.add_argument('--no-install', action='store_true', default=False,
+                    help="Do not install spyder or subrepos")
 parser.add_argument('spyder_options', nargs='*')
 
 args = parser.parse_args()
 
-# Store variable to be used in self.restart (restart spyder instance)
-os.environ['SPYDER_BOOTSTRAP_ARGS'] = str(sys.argv[1:])
-
 assert args.gui in (None, 'pyqt5', 'pyside2'), \
        "Invalid GUI toolkit option '%s'" % args.gui
 
-# Start Spyder with a clean configuration directory for testing purposes
-if args.safe_mode:
-    os.environ['SPYDER_SAFE_MODE'] = 'True'
-
 # Prepare arguments for Spyder's main script
 sys.argv = [sys.argv[0]] + args.spyder_options
 
+# ---- Install sub repos
 
-print("Executing Spyder from source checkout")
-DEVPATH = osp.dirname(osp.abspath(__file__))
+installed_dev_repo = False
+if not args.no_install:
+    for name in REPOS.keys():
+        if not REPOS[name]['editable']:
+            install_repo(name)
+            installed_dev_repo = True
+        else:
+            logger.info("%s installed in editable mode", name)
+if installed_dev_repo:
+    print("\n" + "*" * 79 + "\n" +
+          "\tOne or more repos were installed in develop mode.\n" +
+          "\tPlease rerun bootstrap.\n" +
+          "*" * 79 + "\n")
+    sys.exit()
```

Comment:
```suggestion
    logger.info("Restarting bootstrap to pick up installed subrepos")
    result = subprocess.run([sys.executable, *script_argv, '--no-install'])
    sys.exit(result.returncode)
```

Re-run the script automatically instead of manually